Std/Tactic/Omega/Frontend.lean

```diff
@@ -80,6 +80,18 @@ def mkAtomLinearCombo (e : Expr) : OmegaM (LinearCombo × OmegaM Expr × HashSet
 local macro_rules
   | `(mkAppN $f #[$xs,*]) => (xs.getElems.foldlM (fun x e => `(Expr.app $x $e)) f : MacroM Term)
 
+
+/-- Match on the two ways of spelling successor that are defeq : `n+1`, `n.succ`.
+TODO: support b^(1+n), add a theorem to rewrite b^(1+n) = b^n*b. -/
+def succ? (e : Expr) : Option Expr :=
+  match e.getAppFnArgs with
+  | (``Nat.succ, #[n]) => some n
+  | (``HAdd.hAdd, #[_, _, _, _, a, b]) => do
+     if b == toExpr (1 : Nat)
+     then some a
+     else none
+  | _ => none
```

Comment:
I would have expected a normalization step that rewrites `succ e` into `e + 1`, so that in the match we only have to check for the latter. That way, other rewrites get to automatically make use of it, too

Reply:
Unfortunately, the Omega loop does not seem to have such a step.